.github/workflows/build-and-test.yml

```diff
@@ -9,41 +9,53 @@ jobs:
       matrix:
         include:
           - os: ubuntu
-            os_ver: "22.04"
+            os_ver: "24.04"
             config: Release
             coverage: false
-            cc: gcc-11
-            cxx: g++-11
+            cc: gcc-14
+            cxx: g++-14
+            python_bindings: false
+            arch: x86_64
           - os: ubuntu
-            os_ver: "20.04"
+            os_ver: "22.04"
             config: Release
             coverage: false
             cc: gcc-10
             cxx: g++-10
+            python_bindings: true
+            arch: x86_64
           - os: windows
             os_ver: "2022"
             config: Release
             coverage: false
             cc: cl
             cxx: cl
+            python_bindings: true
+            arch: x86_64
           - os: macos
-            os_ver: "12"
+            os_ver: "14"
```

Comment:
macos-12 is removed.

https://github.com/actions/runner-images/issues/10721

I uses 13 (x86_64) and 14 (arm64, M1)